client/src/connection/ssh/index.ts

```diff
@@ -25,175 +18,199 @@ export interface Config {
+            if (this.config.sasOptions) {
+              resolvedSasOpts.push(...this.config.sasOptions);
+            }
+            const execSasOpts: string = resolvedSasOpts.join(" ");
 
-      stream.write(`${execArgs} ${config.saspath} ${execSasOpts} \n`);
-    });
-  })
-  .on("error", (err) => {
-    reject?.(err);
-    return;
-  });
-
-function getResult() {
-  const runResult: RunResult = {};
-  if (!html5FileName) {
-    resolve?.(runResult);
-    return;
-  }
-  let fileContents = "";
-  conn.exec(`cat ${html5FileName}.htm`, (err: Error, s: ClientChannel) => {
-    if (err) {
-      reject?.(err);
-      return;
-    }
+            this.stream.write(
+              `${execArgs} ${this.config.saspath} ${execSasOpts} \n`
+            );
+          });
+        })
+        .on("error", (err) => {
+          this.clearTimer();
+          this.reject?.(err);
+          return;
+        });
 
-    s.on("data", (data) => {
-      fileContents += data.toString().trimEnd();
-    }).on("close", (code) => {
-      const rc: number = code;
-
-      if (rc === 0) {
-        //Make sure that the html has a valid body
-        //TODO: should this be refactored into a shared location?
-        if (fileContents.search('<*id="IDX*.+">') !== -1) {
-          runResult.html5 = fileContents;
-          runResult.title = "Result";
-        }
-      }
-      resolve?.(runResult);
+      this.setTimer();
+      this.conn.connect(cfg);
     });
-  });
-}
+  };
+  run(code: string, onLog?: (logs: LogLine[]) => void): Promise<RunResult> {
+    this.onLog = onLog;
+    this.html5FileName = "";
+    if (this.logs.length) {
+      this.onLog?.(this.logs.map((line) => ({ type: "normal", line })));
+      this.logs = [];
+    }
 
-function setup(): Promise<void> {
-  return new Promise((pResolve, pReject) => {
-    resolve = pResolve;
-    reject = pReject;
+    return new Promise((_resolve, _reject) => {
+      this.resolve = _resolve;
+      this.reject = _reject;
 
-    if (stream) {
-      resolve();
+      this.stream?.write(`${code}\n`);
+      this.stream?.write(`%put ${endCode};\n`);
+    });
+  }
+  close(): void | Promise<void> {
+    if (!this.stream) {
       return;
     }
-
-    const cfg: ConnectConfig = {
-      host: config.host,
-      port: config.port,
-      username: config.username,
-      privateKey: readFileSync(config.privateKeyPath),
-      readyTimeout: sasLaunchTimeout,
-    };
-    conn.connect(cfg);
-  });
-}
-
-function run(
-  code: string,
-  onLogFn?: (logs: LogLine[]) => void
-): Promise<RunResult> {
-  onLog = onLogFn;
-  html5FileName = "";
-  if (logs.length) {
-    onLog?.(logs.map((line) => ({ type: "normal", line })));
-    logs = [];
+    this.stream.write("endsas;\n");
+    this.stream.end("exit\n");
   }
 
-  return new Promise((_resolve, _reject) => {
-    resolve = _resolve;
-    reject = _reject;
-
-    stream?.write(`${code}\n`);
-    stream?.write(`%put ${endCode};\n`);
-  });
-}
+  setTimer() {
```

Comment:
Can we add method `private`/`protected`/etc to these methods to get a better idea of what's publicly accessible?